src/Compilers/CSharp/Portable/Emitter/EditAndContinue/CSharpSymbolMatcher.cs

```diff
@@ -387,16 +362,54 @@ public override Symbol VisitMethod(MethodSymbol symbol)
 
             public override Symbol VisitModule(ModuleSymbol module)
             {
-                // Only map symbols from source assembly and its previous generations to the other assembly. 
-                // All other symbols should map to themselves.
-                if (module.ContainingAssembly.Identity.Equals(_sourceAssembly.Identity))
+                var otherAssembly = (AssemblySymbol)Visit(module.ContainingAssembly);
+                if (otherAssembly == null)
+                {
+                    return null;
+                }
+
+                // manifest module:
+                if (module.Ordinal == 0)
+                {
+                    return otherAssembly.Modules[0];
+                }
+
+                // match non-manifest module by name:
+                for (int i = 1; i < otherAssembly.Modules.Length; i++)
+                {
+                    var otherModule = otherAssembly.Modules[i];
+                    if (otherModule.Name == module.Name)
```

Comment:
I think we should compare case sensitively. The ECMA spec says 
"The name used in the .module extern directive of the referencing assembly shall
*exactly* match the name used in the .module directive (§II.6.4) of the defining module"

When a module name casing is changed it's a different module name.